utility/serial.cpp

```diff
@@ -92,6 +92,53 @@ void ArdusatSerial::begin(unsigned long baud, bool setXbeeSpeed)
   }
 }
 
+/**
+ * Begin serial communications with a Sparkfun BlueSMiRF module at the specified baud
+ * rate.
+ *
+ * Note that baud rates above ~57600 are not well-supported by SoftwareSerial, and 
+ * even 57600 may cause some bugs.
+ *
+ * @param baud rate for serial communications
+ */
+void ArdusatSerial::beginBluetooth(unsigned long baud)
+{
+  if (_mode == SERIAL_MODE_HARDWARE || _mode == SERIAL_MODE_HARDWARE_AND_SOFTWARE) {
+    bool valid = true;
+    if (baud != 9600) {
+      _soft_serial->end();
+      _soft_serial->begin(9600); // shipping bluesmirf with default of 9600
+      _soft_serial->print("$");  // enter command mode
+      _soft_serial->print("$");
+      _soft_serial->print("$");
+      delay(100);
+
+      // Supported bluesmirf baud rates
+      switch(baud) { // Set Baud rate
+        case 1200: _soft_serial->println("U,1200,N"); break;
+        case 2400: _soft_serial->println("U,2400,N"); break;
+        case 4800: _soft_serial->println("U,4800,N"); break;
+        case 9600: _soft_serial->println("U,9600,N"); break;
+        case 19200: _soft_serial->println("U,192K,N"); break;
+        case 38400: _soft_serial->println("U,384K,N"); break;
+        case 57600: _soft_serial->println("U,576K,N"); break;
+        case 115000: _soft_serial->println("U,115K,N"); break;
```

Comment:
Uno software serial doesn't work above 57600. If you do successfully manage to switch the bluetooth chip into 115000 here, you're effectively bricking it for our purposes (would require the teacher to re-wire to hardware only and/or use a bluetooth config utility and reset manually).

Also, standard ~115k baudrade is 115200, not 115000.